pkg/autodetect/main.go

```diff
@@ -199,24 +202,34 @@ func (b *Background) detectClusterRoles() {
 	}
 }
 
-func (b *Background) cleanDeployments() {
+func (b *Background) cleanDeployments(ctx context.Context) {
 	log.Trace("detecting orphaned deployments.")
 	deployments := &appsv1.DeploymentList{}
 	deployOpts := []client.ListOption{
 		matchingLabelKeys(map[string]string{inject.Label: ""}),
 	}
 
+	namespace, err := k8sutil.GetWatchNamespace()
+	if err != nil {
```

Comment:
Do we need to log the error?

Reply:
Good catch, we could add a message at Trace level.

Reply:
In fact, I think it shouldn't return here, should it?